File: web/serve.py
from pathlib import Path
import json

from starlette.applications import Starlette
from starlette.middleware.gzip import GZipMiddleware
from starlette.responses import JSONResponse
from starlette.routing import Route
from starlette.staticfiles import StaticFiles

# Load the model and maybe shrink it, using top n features
model = json.loads(Path("web/model.json").read_text())

N_FEATURES = 400  # False to use the full model

if N_FEATURES:
    model["features"] = model["features"][:N_FEATURES]
    model["coef"] = [row[:N_FEATURES] for row in model["coef"]]

app = Starlette(
    routes=[
        Route("/model.json", lambda r: JSONResponse(model)),
    ],
)
app.add_middleware(GZipMiddleware)
app.mount("/", StaticFiles(directory="web", html=True))

File: web/serve.js
import express from "express";
import compression from "compression";
import {readFileSync} from "node:fs";

const N_FEATURES = 694; // set to null to use the full model
const model = JSON.parse(readFileSync("model.json", "utf8"));

if (N_FEATURES) {
  model.features = model.features.slice(0, N_FEATURES);
  model.coef = model.coef.map(row => row.slice(0, N_FEATURES));
}

express()
    .use(compression())
    .get("/model.json", (_, res) => res.json(model))
    .use(express.static("web"))
    .listen(8001);
